### ISSUE TITLE
Using fairseq-interative command don't generate only 1 length word,theoritically it  could generate no less than 300 length sentence

### ISSUE THREAD

**Issue by AlexNLP** [facebookresearch/fairseq#4959]
## 🐛 Bug

I have revised the bart,py under the src/models file, and tuned it in the public datasets, but in the inference stage, using fairseq-interative command, the model only generates 1 length word.

### To Reproduce

$ CUDA_VISIBLE_DEVICES=1,2,3,4,5 bash scripts/XXXX/test_constraint.sh data/XXXX/
/home/XX/code/XXXX/src/XXXXX.py:106: UserWarning: __floordiv__ is deprecated, and its behavior will change in a future version of pytorch. It currently rounds toward 0 (like the 'trunc' function NOT 'floor'). This results in incorrect rounding for negative values. To keep the current behavior, use torch.div(a, b, rounding_mode='trunc'), or for actual floor division, use torch.div(a, b, rounding_mode='floor').
  beams_buf = indices_buf // vocab_size
/usr/local/anaconda3/envs/XXX/lib/python3.7/site-packages/fairseq/sequence_generator.py:651: UserWarning: __floordiv__ is deprecated, and its behavior will change in a future version of pytorch. It currently rounds toward 0 (like the 'trunc' function NOT 'floor'). This results in incorrect rounding for negative values. To keep the current behavior, use torch.div(a, b, rounding_mode='trunc'), or for actual floor division, use torch.div(a, b, rounding_mode='floor').
  unfin_idx = idx // beam_size
Wrong format:
Args Namespace(col_header=False, hyp='checkpoints/checkpoint_average_best-3.pt.test_constrained.out.text', row_header=True, table_name=None, tgt='data/XXXX//test.data')
Wrong format: 0 / 54 (0.00%)
E metric:
wandb: WARNING W&B installed but not logged in.  Run `wandb login` or set the WANDB_API_KEY env variable.
Args Namespace(col_header=False, hyp='checkpoints/checkpoint_average_best-3.pt.test_constrained.out.text', metric='E', row_header=True, table_name=None, tgt='data/XXXX//test.data')
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1586/1586 [00:00<00:00, 1606803.42it/s]
Row header: precision = 0.00; recall = 0.00; f1 = 0.00
Non-header cell: precision = 0.00; recall = 0.00; f1 = 0.00
c metric:
wandb: WARNING W&B installed but not logged in.  Run `wandb login` or set the WANDB_API_KEY env variable.
Args Namespace(col_header=False, hyp='checkpoints/checkpoint_average_best-3.pt.test_constrained.out.text', metric='c', row_header=True, table_name=None, tgt='data/XXXX//test.data')
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1586/1586 [00:00<00:00, 1554607.65it/s]
Row header: precision = 0.00; recall = 0.00; f1 = 0.00
Non-header cell: precision = 0.00; recall = 0.00; f1 = 0.00
BS-scaled metric:
wandb: WARNING W&B installed but not logged in.  Run `wandb login` or set the WANDB_API_KEY env variable.
Args Namespace(col_header=False, hyp='checkpoints/checkpoint_average_best-3.pt.test_constrained.out.text', metric='BS-scaled', row_header=True, table_name=None, tgt='data/XXXX//test.data')
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1586/1586 [00:00<00:00, 1628038.70it/s]
Row header: precision = 0.00; recall = 0.00; f1 = 0.00
Non-header cell: precision = 0.00; recall = 0.00; f1 = 0.00

### My attemptance and Expected behavior

I have find the interactive.py in the fairseq_cli file.
```code
            translate_start_time = time.time()
            translations = task.inference_step(
                generator, models, sample, constraints=constraints
            )
```
translations[0][0]['tokens'] has the error torch.Size(2), it should have torch.Size(302).
And I also check the output of bart.py under src/models, the input shape is [batch_size, output_length, num_vocubury], it seems like no problem. I guess the output_layer doesn't take part in the inference stage.

So my problem is how to interactive with the task.inference_step(), which already definite in the tasks/fairseq_task.py
```code
 def inference_step(
        self, generator, models, sample, prefix_tokens=None, constraints=None
    ):
        with torch.no_grad():
            return generator.generate(
                models, sample, prefix_tokens=prefix_tokens, constraints=constraints
            )
```
but I couldn't find the **generator.generate()** function
And the another problem is how do I make sure that the error source from the src/models/bart.py or originates from the inference_step().

### Environment

 - fairseq Version: 0.10.2
 - PyTorch Version: 1.12.1
 - OS: Linux
 - How you installed fairseq: source
 - Python version: 3.7
 - CUDA/cuDNN version: 16.0
 - GPU models and configuration: gtx 3090
